lib/ensime.coffee

```diff
@@ -374,6 +379,34 @@ module.exports = Ensime =
   consumeLinter: (@indieLinterRegistry) ->
 
 
+  provideIntentions: ->
+    getIntentions = (req) =>
+      textEditor = req.textEditor
+      bufferPosition = req.bufferPosition
+      
+      new Promise (resolve) =>
+        @importSuggestions.getImportSuggestions(
+          @clientOfEditor(textEditor),
+          textEditor.getBuffer(),
+          textEditor.getBuffer().characterIndexForPosition(bufferPosition),
+          textEditor.getWordUnderCursor(), # FIXME!
```

Comment:
Hm, yeah, it did a lot that's tied to cursor:

https://github.com/atom/atom/blob/master/src/text-editor.coffee#L1933  

->

https://github.com/atom/atom/blob/master/src/cursor.coffee#L465